pkg/s3/client.go

```diff
@@ -0,0 +1,52 @@
+package s3
+
+import (
+	"github.com/applike/gosoline/pkg/cfg"
+	"github.com/applike/gosoline/pkg/cloud"
+	"github.com/applike/gosoline/pkg/mon"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+	"sync"
+)
+
+//go:generate mockery --name Client
+type Client interface {
+	s3iface.S3API
+}
+
+var c = struct {
+	sync.Mutex
+	instance map[string]s3iface.S3API
+}{}
+
+type Settings struct {
+	BasePath string
+	BucketName string
+	Client cloud.ClientSettings
+}
+
+func ProvideClient(config cfg.Config, logger mon.Logger, settings *Settings) s3iface.S3API {
```

Comment:
can we move the s3 client to `gosoline/pkg/cloud/aws_clients.go` ?

Reply:
Why? This is following the project layering. Basically the client here is S3 specific. So no, I would leave it here, no?

Reply:
The sns and sqs stuff will also be moved to a more aws related space in the future. Most likely be done while switching to the aws sdk v2. Furthermore i would like to keep the root packages free of any cloud provider specific packages and rather group them in the cloud package.

Reply:
okay, moving the s3 folder form `pkg` to `pkg/cloud/aws` should be right to do then right?

Reply:
yes, that would also be fine :+1:

Reply:
And that is done 👍